code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,1227 @@
+	max_cooldown = 100
+	///used to remember the mobs currently contained by the rune
+	var/list/mob_list
+
+
+/obj/effect/warped_rune/goldspace/Initialize()
+	. = ..()
+	cooldown = 0 //the cooldown is only used when someone step on the rune
+
+
+///Will vore hostile mobs on the rune and add them to the mob_list of the rune. 5 maximum number of mobs in the rune.
+/obj/effect/warped_rune/goldspace/attack_hand(mob/living/user)
+	if(cooldown > world.time)
+		to_chat(user,"<span class='notice'>The rune needs a little bit more time before absorbing more animals!</span>")
+		return
+
+	if(length(mob_list) >= 5) //no more than 5 good boys allowed.
+		to_chat(user,"<span class='notice'>The rune's army is as full as it can be!</span>")
+		return
+
+	for(var/mob/living/simple_animal/hostile/captured in rune_turf)
+		if(istype(captured,/mob/living/simple_animal/hostile/megafauna) || captured.stat == DEAD) //Megafaunas are too thick to be proper soldiers
+			return
+
+		LAZYADD(mob_list, captured.type)
+		qdel(captured)
+		playsound(rune_turf, 'sound/effects/splat.ogg', 20, TRUE)
+		to_chat(user,"<span class='warning'>The rune somehow absorbs [captured] into itself!</span>")
+
+
+///will unleash all the mobs in the rune on the rune and give them the person that just crossed the rune their first target.
+/obj/effect/warped_rune/goldspace/Crossed(atom/movable/crossing)
+	. = ..()
+	if(!istype(crossing,/mob/living/carbon/human))
+		return
+
+	for(var/mob_type in mob_list)
+		var/mob/living/simple_animal/hostile/spawned_mob = new mob_type(rune_turf)
+		spawned_mob.move_to_delay = 1 //make them much faster to make them even more lethal. 1 is slightly slower than bees
+		spawned_mob.GiveTarget(crossing)//Because they are given a target from the get go they will react much faster after spawn than usual.
+		LAZYREMOVE(mob_list, mob_type)
+		cooldown = world.time + max_cooldown //you can't instantly reuse the rune to send the mobs back into the rune.
+
+
+/obj/effect/warped_rune/goldspace/Destroy()
+	mob_list = null
+	return ..()
+
+
+/*Adamantine rune, will spawn ores depending on the mineral rocks surrounding it. Here to make miners do their job even less.  */
+
+
+/obj/item/slimecross/warping/adamantine
+	colour = "adamantine"
+	runepath = /obj/effect/warped_rune/adamantinespace
+	effect_desc = "draws a rune capable of copying the ores of nearby mineral rocks."
+
+
+/obj/effect/warped_rune/adamantinespace  //doesn't have a rune icon yet please spriters help me I can't sprite for shit I beg you
+	desc = "The universe's ressource are nothing but tools for us to use and abuse."
+	max_cooldown = 300  //"mines" things every 30 seconds.
+	icon_state = "mining_rune"
+
+
+/obj/effect/warped_rune/adamantinespace/Initialize()
+	. = ..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/adamantinespace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		auto_mining()
```

Comment:
Does this need to be a continuous process instead of activated on demand?
Unless this is moving, it won't need to scan its surroundings after the first time, and if it's moving the user can keep activating it as needed, no?